scripts/ci/update-version-list/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/thoas/go-funk"
+
+	"github.com/ActiveState/cli/internal/condition"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/httputil"
+)
+
+// Where the master version file lives on S3.
+const S3PrefixURL = "https://state-tool.s3.amazonaws.com/"
+const S3Bucket = "update/state/"
+const VersionsJson = "versions.json"
+
+// Valid channels to update the master version file with.
+var ValidChannels = []string{"beta", "release", "LTS"}
+
+func init() {
+	if !condition.OnCI() {
+		// Allow testing with artifacts produced by `state run generate-test-update`
+		ValidChannels = append(ValidChannels, "test-channel")
+	}
+}
+
+func main() {
+	if len(os.Args) != 2 {
+		log.Fatalf("Usage: %s <build-dir>", os.Args[0])
+	}
+
+	// Fetch the current master list from S3.
+	versions := []map[string]string{}
+	fmt.Printf("Fetching master %s file from S3\n", VersionsJson)
+	bytes, err := httputil.Get(S3PrefixURL + S3Bucket + VersionsJson)
+	if err != nil {
+		log.Fatalf("Failed to fetch file: %s", err.Error())
+	}
+	err = json.Unmarshal(bytes, &versions)
+	if err != nil {
+		log.Fatalf("Failed to decode JSON: %s", err.Error())
+	}
+
+	// Find info.json files to add to the master list and add them.
+	updated := false
+	buildDir := os.Args[1]
+	fmt.Printf("Searching for info.json files in %s\n", buildDir)
+	files, err := fileutils.ListDir(buildDir, false)
+	if err != nil {
+		log.Fatalf("Failed to search %s: %s", buildDir, err.Error())
+	}
+	for _, file := range files {
+		if file.Name() != "info.json" {
+			continue
+		}
+		channel := strings.Split(file.RelativePath(), string(filepath.Separator))[0]
+		if !funk.Contains(ValidChannels, channel) {
+			continue
+		}
+		fmt.Printf("Found %s\n", file.RelativePath())
+		bytes, err := fileutils.ReadFile(file.Path())
+		if err != nil {
+			log.Fatalf("Unable to read file: %s", err.Error())
+		}
+		info := map[string]string{}
+		err = json.Unmarshal(bytes, &info)
```

Comment:
We should link the types used to generate and use the info so we don't accidentally break things in the future. Also to eliminate unmarshalling to a key/value map; I'd like to rely on Go's type system.